### PR TITLE
Mantine Checkbox

### LINKED ISSUE #32270: Checkbox component
Docs

- [Product doc](https://www.notion.so/metabase/Design-System-Checkbox-component-8a916dfc82e64454981e51eba8978cef?pvs=4)
- [Figma component](https://www.figma.com/file/sF1qSHk6yVqO1rFgmH0nzT/Input-%2F-Checkbox?type=design&node-id=5-70&mode=design&t=TqHJM8JVRenem5Mk-11)

```[tasklist]
### Tasks
- [ ] Export the [Mantine](https://mantine.dev/core/checkbox/) `Checkbox` component from the `metabase/frontend/src/ui` according to the component’s category.
- [ ] Make sure the component style matches the component in Figma. You can add component specific styling to the `components:` section of `frontend/src/metabase/ui/theme.ts`
- [ ] Create a storybook file for the new component in the same directory. You can use `frontend/src/metabase/ui/components/inputs/Radio/Radio.stories.mdx` as an example.

```

### PR DESCRIPTION
**Pull request by oisincoveney**
Closes https://github.com/metabase/metabase/issues/32270

### Description

Adds styles and stories to the Mantine Checkbox and Checkbox group, based off the Figma file in the linked issue. I tried to follow the borders/spacing/etc as closely as possible - if there is some overkill in the code then I'm happy to simplify it.

### How to verify

Look at the Storybook for `Checkbox and Checkbox Groups` and see that the checkbox styling matches Figma. I'm happy to add more stories if necessary.

### Checklist

- [X] Stories have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/ui/components/inputs/Checkbox/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { Checkbox } from "@mantine/core";
```
Comment (kdoh) [inline frontend/src/metabase/ui/components/inputs/Checkbox/index.ts:1]: We'll need to set a default prop for the checkbox to use our icon component somehow. Any ideas on the best pattern for that? https://mantine.dev/core/checkbox/#replace-icon has some related info.

**Review by kdoh — CHANGES_REQUESTED** [review]
Overall a great start. Biggest thing to solve is just how to make sure this comes with our proper check icon out of the box. Once that's working I'll take a slightly closer look at spacing and colors with fresh eyes.

File: frontend/src/metabase/ui/components/inputs/Checkbox/index.ts (hunk 1 above)
Comment (oisincoveney) [reply frontend/src/metabase/ui/components/inputs/Checkbox/index.ts:1]: Ok sounds good - I wasn't sure if we were going to use our own icons for the Mantine stuff just yet, the default icon makes me super uncomfortable 😅

File: frontend/src/metabase/ui/theme.ts [hunk 2]
```diff
@@ -45,6 +46,71 @@ export const theme: MantineThemeOverride = {
         };
       },
     },
+    Checkbox: {
+      defaultProps: {
+        icon: CheckboxIcon,
```
Comment (oisincoveney) [inline frontend/src/metabase/ui/theme.ts:51]: We can define default props for all Mantine components

**Review by kdoh — APPROVED** [review]
LGTM

**Pull request merged**